From 13a14579cc1bb772735f895dd5b4b90c6812f3ee Mon Sep 17 00:00:00 2001
From: Chris Feng <chris.w.feng@gmail.com>
Date: Sun, 29 Jul 2018 00:00:00 +0000
Subject: [PATCH] Fix issues with destroying full screen X windows

* exwm-manage.el (exwm-manage--unmanage-window): Set the Emacs window
of an full screen X window as non-dedicated before killing its buffer
so as not to cause other side effects.
---
 exwm-manage.el | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/exwm-manage.el b/exwm-manage.el
index a0a9e05..349157f 100644
--- a/exwm-manage.el
+++ b/exwm-manage.el
@@ -392,6 +392,10 @@ manager is shutting down."
                                :window window :parent exwm--root :x 0 :y 0))
             (xcb:+request exwm--connection
                 (make-instance 'xcb:DestroyWindow :window container))))
+        (when (exwm-layout--fullscreen-p)
+          (let ((window (get-buffer-window)))
+            (when window
+              (set-window-dedicated-p window nil))))
         (exwm-manage--set-client-list)
         (xcb:flush exwm--connection))
       (let ((kill-buffer-func
